app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -153,9 +198,27 @@ export class CognitoImpl implements Cognito {
 
     // === Interface `impl`s ===
 
+    /** We want to signal to Amplify to fire a "custom state change" event when the user is
+     * redirected back to the application after signing in via an external identity provider. This
+     * is done so we get a chance to fix the location history that Amplify messes up when it
+     * redirects the user to the identity provider's authentication page.
+     *
+     * In order to do so, we need to pass custom state along for the entire OAuth flow, which is
+     * obtained by calling this function. This function will return the current location path if
+     * the user is signing in from the desktop application, and `undefined` otherwise.
```

Comment:
`undefined` should never be used as a created value in JS. `undefined` means not defined. If you want to show lack of value, use `null` instead.

Reply:
should be done as of latest commit